### PR TITLE
Log incoming HTTP requests (analytics) and add domain vision statement (docs)

### PR DESCRIPTION
**Pull request by dwinston**
kicked off by https://github.com/microbiomedata/nmdc-schema/pull/85#issuecomment-1737749889

### CODE REVIEW

**Review by eecavanna — APPROVED** [review]
Cool concept and approach.

File: nmdc_runtime/api/analytics.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+from starlette.types import ASGIApp
+from toolz import merge
+
+from nmdc_runtime.api.db.mongo import get_mongo_db
+
+_requests = []
+_last_posted = datetime.now()
+
+
+def _post_requests(collection: str, requests_data: List[Dict], source: str):
```
Comment (eecavanna) [inline nmdc_runtime/api/analytics.py:23]: Suggestion: Instead of "post", maybe "save" or "write". I don't think this warrants blocking a merge, though.

**Comment by turbomam** [comment]
@eecavanna is in a better position to address the implementation than I am

I don't see yet how I would use this. That doesn't concern me, but I would like some clarification about the kinds of usage that will become eligible for analytics.

Is this for requests like
- /nmdcschema/{collection_name}/{doc_id}
- /nmdcschema/{collection_name}

or is it for interactions with our ga4gh DRS system?

Maybe I wan't clear about my use case from #85. At one point, we had links on the nmdc-schema repo's README.md that (if I remember correctly) enabled downloading of our metadata in bulk, from the DRS system. 

https://polyneme.nyc3.cdn.digitaloceanspaces.com/nmdc/mongo/export/dwinston_share/biosample_set.json.bz2

That is no longer exposed in the README.md, and I'm mainly trying to understand if that was  bad decision on my part. So I thought this PR was going to make it easier to track who was using the DRS system.

I really don't understand DRS, so maybe I'm asking unhelpful questions.

**Comment by eecavanna** [comment]
Hi @dwinston, this branch—including the latest commit—looks good to me. I'm OK with it being merged in.

I propose renaming the PR:

```diff
- add light api analytics for monitoring/observability 
+ Log incoming HTTP requests (analytics) and add domain vision statement (docs)
```

**Review by eecavanna — APPROVED** [review]

**Pull request merged**